azure.yaml

```diff
@@ -1,34 +1,80 @@
-# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
+trigger:
+- main
 
-name: azure-search-openai-demo
-metadata:
-  template: azure-search-openai-demo@0.0.2-beta
-services:
-  backend:
-    project: ./app/backend
-    language: java
-    host: appservice
-    hooks:
-      prepackage:
-        windows:
-          shell: pwsh
-          run:  cd ../frontend;npm install;npm run build
-          interactive: true
-          continueOnError: false
-        posix:
-          shell: sh
-          run:  cd ../frontend;npm install;npm run build
-          interactive: true
-          continueOnError: false
-hooks:
-    postprovision:
-      windows:
-        shell: pwsh
-        run: ./scripts/prepdocs.ps1
-        interactive: true
-        continueOnError: false
-      posix:
-        shell: sh
-        run: ./scripts/prepdocs.sh
-        interactive: true
-        continueOnError: false
+pool:
+  vmImage: 'ubuntu-latest'
+
+jobs:

```

Comment:
would be better to separate build and deploy using stages like below:
stages:
  - stage: Build
    displayName: Build Stage
    jobs:
      - job: Build
  - stage: DeployDev
  displayName: Deploy on continuous integration environment
  jobs:
    - deployment:
      environment: dev
      displayName: 'Deploy on dev enviroment'

Reply:
In reality we wouldn't use build and deploy in the same pipeline.  Release has it's own thing in AzDO.  But you can't really have a yaml or text output of the deploy to store in repo

Reply:
The yaml multi staging pipeline is able to manage the E2E deployment pipeline from dev to prod (also including manual approval process). the Release is just the (old) UI based way of doing that. As we are trying to follow DevOps "pipeline as code" best practice here I still thing this is the right path